types/config.go

```diff
@@ -32,7 +33,7 @@ type ConfigDetails struct {
 
 // LookupEnv provides a lookup function for environment variables
 func (cd ConfigDetails) LookupEnv(key string) (string, bool) {
-	v, ok := cd.Environment[key]
+	v, ok := utils.EnvResolver(cd.Environment)(key)
```

Comment:
as `EnvResolver` is not used in any other place, I'd prefer we keep things simple with `EnvResolverWithCase` logic being directly included here